src/modules/hip/kernel/tensor_sum.hpp

```diff
@@ -0,0 +1,1439 @@
+    partialBSumSharedRowPtr[hipThreadIdx_x] = (src_ui24.ui8[2].ui1[0] +
+                                               src_ui24.ui8[2].ui1[1] +
+                                               src_ui24.ui8[2].ui1[2] +
+                                               src_ui24.ui8[2].ui1[3]);                          // perform small work of reducing B uint4s to uint using 16 x 16 threads and store in Shared
+
+    __syncthreads();                                                                             // syncthreads after Shared load
+
+    // Reduction of 16 uints on 16 threads per block in x dimension (for every y dimension)
+    for (int threadMax = 8; threadMax >= 1; threadMax /= 2)
+    {
+        if (hipThreadIdx_x < threadMax)
+        {
+            partialRSumSharedRowPtr[hipThreadIdx_x] += partialRSumSharedRowPtr[hipThreadIdx_x + threadMax];
+            partialGSumSharedRowPtr[hipThreadIdx_x] += partialGSumSharedRowPtr[hipThreadIdx_x + threadMax];
+            partialBSumSharedRowPtr[hipThreadIdx_x] += partialBSumSharedRowPtr[hipThreadIdx_x + threadMax];
+        }
+        __syncthreads();
+    }
+
+    if (hipThreadIdx_x == 0)
+    {
+        // Reduction of 16 uints on 16 threads per block in y dimension
+        for (int threadMax = 8, increment = 128; threadMax >= 1; threadMax /= 2, increment /= 2)
+        {
+            if (hipThreadIdx_y < threadMax)
+            {
+                partialRSumSharedRowPtr[0] += partialRSumSharedRowPtr[increment];
+                partialGSumSharedRowPtr[0] += partialGSumSharedRowPtr[increment];
+                partialBSumSharedRowPtr[0] += partialBSumSharedRowPtr[increment];
+            }
+            __syncthreads();
+        }
+
+        // Final store to dst
+        if (hipThreadIdx_y == 0)
+        {
+            int idx = ((hipBlockIdx_z * hipGridDim_y + hipBlockIdx_y) * hipGridDim_x + hipBlockIdx_x) * 3;
+            tensorSumArr[idx] = partialRSumSharedRowPtr[0];
+            tensorSumArr[idx + 1] = partialGSumSharedRowPtr[0];
+            tensorSumArr[idx + 2] = partialBSumSharedRowPtr[0];
+        }
+    }
+}
+
+//Handle I8
+__global__ void tensor_sum_pln3(Rpp8s *srcPtr,
+                                uint3 srcStridesNCH,
+                                Rpp32s *tensorSumArr,
+                                RpptROIPtr roiTensorPtrSrc)
+{
+    int id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x) * 8;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    __shared__ int partialRSumShared[16][16];                                                   // 16 rows of src, 128 reduced cols of src in a 16 x 16 thread block
+    __shared__ int partialGSumShared[16][16];
+    __shared__ int partialBSumShared[16][16];
+    int *partialRSumSharedRowPtr = &partialRSumShared[hipThreadIdx_y][0];                       // int pointer to beginning of each row in Shared
```

Comment:
Shared should ideally be the last word in any of our variables so we can convert to _smem